_posts/2021-01-22-x-ways-to-write-meaningful-commit-messages.md

```diff
@@ -0,0 +1,140 @@
+
+If you commit two changes together, for example, a bug fix, and a minor refactoring, It might not cause a very long commit message, but it can cause some problems. Let's say the bug fix made some other bugs. In that case, you should roll back to your recent commit. It results in the loss of that refactoring as well. It's not efficient, and It's not atomic.
+
+### Short and Unambiguous:
+The message should describe what changes your code makes to the base code. It should answer the question: **"What happens if the changes are applied?"**. If you can not be short, it might be because the commit is not atomic, and it's too much change in one commit.
+
+#### Active Voice:
+ Use the **imperative, present tense**. It is easier to read and skim.
+> **Right**: Add feature for a user to upload pictures
+>
+> **Wrong**: Added feature ... (passive voice, past tense)
+
+#### Detailed Enough:
+Super-detailed commit messages are frustrated as well. You can find that much detail in the system. For example, If your version control is Git, you can see all the changed files in Git. So instead of answering "what are the changes? ", it's better to answer **"What are the changes for?"**.
+
+### Formatting:
+Bellow, we describe a simplified format of what Angular suggests.
+This format consists of a header, a body, and a footer with a blank line between each section.
+
+Header: Mandatory - Up to 50 characters (as Git suggests)
+
+> (type) (scope): (short summary)
+
+Description: Mandatory (except for docs) - at least 20 characters
+> (body)
+
+Footer: Optional
+> (footer)
+
+The header consists of a type and a summary part. Some add scope in between. But since it's optional, we only discuss it briefly.
+
+#### Type:
+The type of commit message is that change made for a particular problem. For example, if you've fixed a bug or add a feature, or maybe changed something related to the docs, the type would be Fix, Add, or Doc. It could be whatever you and your teammates decide to be as long as it's understandable and consistent. It's better to be capital.
+
+#### Scope:
+It's the package or module that is affected by the change. As mentioned before, it's optional.
+
+#### Summary:
+As Angular suggests: "It should be present tense. Not capitalized. No period in the end.", and **imperative like the type**.
+As [Chris Beams](https://chris.beams.io/posts/git-commit/) mentions here, the summary should always be able to complete the following sentence: 
+
+If applied, this commit will ...
+
+**If applied, this commit will** 
+add authorization for document access
+
+Bellow, part "Fix" is the type, and the sentence after ":" is the summary.
+> **Right**: Fix: add authorization for document access
+> 
+> **Wrong**: Fix: Add authorization for document access (Capitalized)
+>
+> **Wrong**: Fix: added authorization for document access (Not present tense)
+>
+> **Wrong**: Fix: add authorization for document access. (Used period in the end)
```

Comment:
It would be nice to describe somewhere WHY capitalized is better for the type and WHY not capitalized is better for the summary and WHY present tense and active voice are best. That's something that many other articles are lacking. It need not necessarily be explained at this point in the text, though.

Reply:
Thank you for your feedback. This is something I've struggled with it too. Great! I do it.